orc8r/cloud/go/services/orchestrator/servicers/builder_servicer_test.go

```diff
@@ -311,12 +340,162 @@ func TestBaseOrchestratorMconfigBuilder_Build(t *testing.T) {
 				LogLevel:       protos.LogLevel_INFO,
 				EventVerbosity: -1,
 			},
+			"state": &mconfig_protos.State{
+				//expected value differs from map due to added jitter
+				SyncInterval: 68,
+			},
+		}
+
+		actual, err := buildBaseOrchestrator(&nw, &graph, "gw1")
+		assert.NoError(t, err)
+		assert.Equal(t, expected, actual)
+	})
+	// No state config added to Network Configs
+	t.Run("no state config", func(t *testing.T) {
+		nw := configurator.Network{ID: "n1"}
+		gw := configurator.NetworkEntity{
+			Type: orc8r.MagmadGatewayType,
+			Key:  "gw1",
+			Config: &models.MagmadGatewayConfigs{
+				AutoupgradeEnabled:      swag.Bool(true),
+				AutoupgradePollInterval: 300,
+				CheckinInterval:         60,
+				CheckinTimeout:          10,
+				DynamicServices:         []string{},
+				FeatureFlags:            map[string]bool{},
+			},
+		}
+		tier := configurator.NetworkEntity{
+			Type: orc8r.UpgradeTierEntityType,
+			Key:  "default",
+			Config: &models.Tier{
+				Name:    "default",
+				Version: "1.0.0-0",
+				Images: []*models.TierImage{
+					{Name: swag.String("Image1"), Order: swag.Int64(42)},
+					{Name: swag.String("Image2"), Order: swag.Int64(1)},
+				},
+			},
+		}
+		graph := configurator.EntityGraph{
+			Entities: []configurator.NetworkEntity{gw, tier},
+			Edges: []configurator.GraphEdge{
+				{From: tier.GetTypeAndKey(), To: gw.GetTypeAndKey()},
+			},
+		}
+
+		expected := map[string]proto.Message{
+			"control_proxy": &mconfig_protos.ControlProxy{LogLevel: protos.LogLevel_INFO},
+			"magmad": &mconfig_protos.MagmaD{
+				LogLevel:                protos.LogLevel_INFO,
+				CheckinInterval:         60,
+				CheckinTimeout:          10,
+				AutoupgradeEnabled:      true,
+				AutoupgradePollInterval: 300,
+				PackageVersion:          "1.0.0-0",
+				Images: []*mconfig_protos.ImageSpec{
+					{Name: "Image1", Order: 42},
+					{Name: "Image2", Order: 1},
+				},
+				DynamicServices: nil,
+				FeatureFlags:    nil,
+			},
+			"metricsd": &mconfig_protos.MetricsD{LogLevel: protos.LogLevel_INFO},
+			"td-agent-bit": &mconfig_protos.FluentBit{
+				ExtraTags:        map[string]string{"network_id": "n1", "gateway_id": "gw1"},
+				ThrottleRate:     1000,
+				ThrottleWindow:   5,
+				ThrottleInterval: "1m",
+			},
+			"eventd": &mconfig_protos.EventD{
+				LogLevel:       protos.LogLevel_INFO,
+				EventVerbosity: -1,
+			},
+			"state": &mconfig_protos.State{
+				SyncInterval: 68,
+			},
 		}
 
 		actual, err := buildBaseOrchestrator(&nw, &graph, "gw1")
 		assert.NoError(t, err)
 		assert.Equal(t, expected, actual)
 	})
+
+	// test sync interval jitter
```

Comment:
Nit: sentence-case your comments

Reply:
How is this test different from the previous tests, with regard to testing sync interval jitter? The network id actually needs to be different (?)

Reply:
the jitter is based on the gateway id not the network id

Reply:
Ah of course. Okay in any case, ensure we have tests that cover different gateway IDs such that they cause different jittered values

Reply:
they do - to make it clear at the top of the test i have 2 diff expected sync intervals based on different gateway ids

Reply:
👍💯